addon/test-helper/deprecation.js

```diff
@@ -1,53 +1,50 @@
-/* globals QUnit */
-
-import MethodCallExpectation from "./method-call-expectation";
-
-var NONE = function(){};
+import MethodCallExpectation from './method-call-expectation';
+import MethodCallTracker from './method-call-tracker';
 
 var DeprecationAssert = function(env){
   this.env = env;
-
   this.reset();
 };
 
 DeprecationAssert.prototype = {
-
   reset: function(){
-    this.expecteds = null;
-    this.actuals = null;
+    this.tracker = new MethodCallTracker(this.env.Ember, 'deprecate');
```

Comment:
It is a little funky that `this.tracker` is used here, but `invokeWithNewContext` could be called with a completely different tracker than this one.

Can everything use the tracker on `this` for the sake of not having two floating around when one is all that is necessary?